test/functional/connection.test.js

```diff
@@ -273,4 +274,31 @@ describe('Connection', function () {
       done();
     }
   });
+
+  it('should be able to connect again after close', function () {
+    return withClient.call(this, (client, done) => {
```

Comment:
I'm using this instead of just `withClient((client, done) ...` because I couldn't see test output (warning messages) without this change.

Reply:
@HanaPearlman Yes! this is super confusing, there's ticket in the back log to address this I added a comment in there about how to use this as I believe it was intended. [NODE-2764](https://jira.mongodb.org/browse/NODE-2764?focusedCommentId=3444002&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3444002)

Reply:
Whoops, took a closer look at `withClient` and it seems like the right approach is to bind `withClient` to `this`.

Reply:
@reggi Thanks for that! Happy to make that change now to replace the test callback or keep the `bind` syntax; not 100% clear on what is best...